tests/unit_tests/test_lobbyconnection.py

```diff
@@ -501,194 +500,20 @@ async def test_command_admin_closelobby(mocker, lobbyconnection):
-
-    assert previous_ban is not None
-
-    await lobbyconnection.on_message_received({
-        'command': 'admin',
-        'action': 'closelobby',
-        'user_id': banme.id,
-        'ban': {
-            'reason': 'Unit test - already banned',
-            'duration': 1000
-        }
-    })
-
-    async with database.acquire() as conn:
-        result = await conn.execute(select([ban.c.id]).where(ban.c.player_id == banme.id))
-
-        bans = [row['id'] async for row in result]
-
-    assert len(bans) == 1
-    assert bans[0] == previous_ban["id"]
-
-
-async def test_command_admin_closelobby_with_ban_duration_no_period(mocker, lobbyconnection, database):
-    player = mocker.patch.object(lobbyconnection, 'player')
-    player.login = 'Sheeo'
+async def test_command_admin_closeFA(lobbyconnection, player_factory):
+    player = lobbyconnection.player
     player.id = 1
-    player.admin = True
-    banme = mock.Mock()
-    banme.id = 200
-    banme.lobby_connection = asynctest.create_autospec(LobbyConnection)
-    lobbyconnection.player_service = {1: player, banme.id: banme}
-
-    mocker.patch('server.lobbyconnection.func.now', return_value=text('FROM_UNIXTIME(1000)'))
-    await lobbyconnection.on_message_received({
-        'command': 'admin',
-        'action': 'closelobby',
-        'user_id': banme.id,
-        'ban': {
-            'reason': 'Unit test - ban duration',
-            'duration': 3600*24
-        }
-    })
-
-    banme.lobby_connection.kick.assert_any_call()
-
-    async with database.acquire() as conn:
-        result = await conn.execute(select([ban.c.expires_at]).where(ban.c.player_id == banme.id))
-
-        bans = [row['expires_at'] async for row in result]
-
-    assert len(bans) == 1
-    assert bans[0] == datetime.utcfromtimestamp(3600*24 + 1000)
-
-
-async def test_command_admin_closelobby_with_ban_bad_period(mocker, lobbyconnection, database):
-    player = mocker.patch.object(lobbyconnection, 'player')
-    player.admin = True
-    banme = mock.Mock()
-    banme.id = 1
-    lobbyconnection.player_service = {1: player, banme.id: banme}
-
-    await lobbyconnection.on_message_received({
-        'command': 'admin',
-        'action': 'closelobby',
-        'user_id': banme.id,
-        'ban': {
-            'reason': 'Unit test',
-            'duration': 2,
-            'period': ') injected!'
-        }
-    })
-
-    banme.lobbyconnection.kick.assert_not_called()
-    lobbyconnection.protocol.send_message.assert_called_once_with({
-        'command': 'notice',
-        'style': 'error',
-        'text': "Period ') INJECTED!' is not allowed!"
-    })
-
-    async with database.acquire() as conn:
-        result = await conn.execute(select([ban]).where(ban.c.player_id == banme.id))
-
-        bans = [row['reason'] async for row in result]
-
-    assert len(bans) == 0
-
-
-async def test_command_admin_closelobby_with_ban_injection(mocker, lobbyconnection, database):
```

Comment:
Do I even want to know why this test was necessary at some point? :sweat_smile:

Reply:
You can look at the deleted code, it was using string formatting in the `text` call because mysql has no way of parametrizing interval durations.